### ISSUE TITLE
Ignore "used before defined" with forwardRef

### ISSUE THREAD

**Issue by grosch** [eslint/eslint#13266]
**What rule do you want to change?**

@typescript-eslint/no-use-before-define

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

Ideally default behavior, but an option is an option :)

**Please provide some example code that this change will affect:**

```js
@Component({
    providers: [{
        provide: NG_VALUE_ACCESSOR,
        useExisting: forwardRef(() => MyComponent)
        multi: true
    }]
})
export class MyComponent implements ControlValueAccessor {
```

**What does the rule currently do for this code?**

It's complaining about MyComponent being used before defined, but this syntax is valid and expected here with the forwardRef.

**What will the rule do after it's changed?**

Not complain about this rule when inside a forwardRef like this.

**Are you willing to submit a pull request to implement this change?**

Sorry but I don't have that skillset.

**Comment by g-plane** [comment]
@grosch Hello, this rule isn't in ESLint core. You can create issue at [here](https://github.com/typescript-eslint/typescript-eslint/issues).
